waku/node/waku_node.nim

```diff
@@ -673,25 +675,45 @@ proc filterUnsubscribeAll*(
 
 ## Waku archive
 proc mountArchive*(
-    node: WakuNode, driver: ArchiveDriver, retentionPolicy = none(RetentionPolicy)
+    node: WakuNode,
+    driver: waku_archive.ArchiveDriver,
+    retentionPolicy = none(waku_archive.RetentionPolicy),
 ): Result[void, string] =
-  node.wakuArchive = WakuArchive.new(driver = driver, retentionPolicy = retentionPolicy).valueOr:
+  node.wakuArchive = waku_archive.WakuArchive.new(
+    driver = driver, retentionPolicy = retentionPolicy
+  ).valueOr:
     return err("error in mountArchive: " & error)
 
   node.wakuArchive.start()
 
   return ok()
 
+proc mountLegacyArchive*(
+    node: WakuNode,
+    driver: waku_archive_legacy.ArchiveDriver,
+    retentionPolicy = none(waku_archive_legacy.RetentionPolicy),
+): Result[void, string] =
+  node.wakuLegacyArchive = waku_archive_legacy.WakuArchive.new(
+    driver = driver, retentionPolicy = retentionPolicy
+  ).valueOr:
+    return err("error in mountArchive: " & error)
+
+  node.wakuLegacyArchive.start()
+
+  return ok()
+
 ## Legacy Waku Store
 
 # TODO: Review this mapping logic. Maybe, move it to the appplication code
-proc toArchiveQuery(request: legacy_store_common.HistoryQuery): ArchiveQuery =
-  ArchiveQuery(
+proc toArchiveQuery(
```

Comment:
Notice that it is being used in both legacy (`mountLegacyStore`) and non-legacy (`mountStore`). I don't know where this would cause an issue.

Reply:
It's changed in another PR no worry.